polaris-service/src/main/java/io/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -897,10 +918,11 @@ private void validateLocationForTableLike(
           // }
         },
         () -> {
-          if (location.startsWith("file:") || location.startsWith("http")) {
+          if (locations.stream()
+              .anyMatch(location -> location.startsWith("file:") || location.startsWith("http"))) {
             throw new ForbiddenException(
-                "Invalid location '%s' for identifier '%s': File locations are not allowed",
-                location, identifier);
+                "Invalid locations '%s' for identifier '%s': File locations are not allowed",
```

Comment:
This is going to be a little confusing because we are reporting that all locations are invalid when we are only sure that at least 1 is. We should probably do a filter so we can report only those that are violating the perquisites.

Reply:
Yeah, this was me being lazy :)